app/uk/gov/hmrc/gform/submission/PdfAndXmlSummariesFactory.scala

```diff
@@ -172,10 +173,46 @@ object PdfAndXmlSummariesFactory {
             )
           ).map(xml => compact(JsonMethods.render(org.json4s.Xml.toJson(xml))))
         case DataOutputFormat.HBS =>
-          val focussedTree = FocussedHandlebarsModelTree(modelTree, modelTree.value.model)
           hmrcDms.payload.fold(
-            throw new RuntimeException(s"Payload is empty in the destination '${formTemplate._id}-${hmrcDms.id.id}'.")
-          )(payload => Some(RealHandlebarsTemplateProcessor(payload, accumulatedModel, focussedTree, TemplateType.XML)))
+            throw new RuntimeException(s"Payload is empty in the destination '$formTemplateId-${hmrcDms.id.id}'.")
+          ) { payload =>
+            val focussedTree: FocussedHandlebarsModelTree =
+              FocussedHandlebarsModelTree(modelTree, modelTree.value.model)
+            val filledHandlebarTemplate: String =
+              RealHandlebarsTemplateProcessor(payload, accumulatedModel, focussedTree, TemplateType.XML)
+
+            Some(
+              RoboticsXMLGenerator(
+                formTemplateId,
+                hmrcDms.dmsFormId,
+                submissionRef,
+                now,
+                l,
+                envelopeId,
+                filledHandlebarTemplate
+              )
+            ).map(formatHandlebarWithXmlElements)
+          }
+      }
+
+    private def formatHandlebarWithXmlElements(handlebarWithRoboticsElements: NodeSeq): String = {
+      val formattedHandlebar = handlebarWithRoboticsElements.toString().replace("><", ">\n<")
+
+      val indentedHandlebar = wrapAndIndentHandlebarXml(None, formattedHandlebar)
+
+      XmlGeneratorService.xmlDec + "\n" + wrapAndIndentHandlebarXml(
+        Some("<data xmlns:xfa=\"http://www.xfa.org/schema/xfa-data/1.0/\">"),
+        indentedHandlebar
+      )
+    }
+
+    private def wrapAndIndentHandlebarXml(wrappingTag: Option[String], contents: String): String =
```

Comment:
would it be better to use `scala.xml.PrettyPrinter`  for xml formatting?

Reply:
I was looking into this but apparently it's not thread safe. If that doesn't matter then I'll be happy to make the change to use it

Reply:
After discussion, will change to use `PrettyPrinter`